### PR TITLE
Allow empty SMTP user/pass

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Don't create the smtp.auth entry if no user/pass provided. This allows support for SMTP servers that do not require authentication.

## Related Issue(s)

https://github.com/flowforge/flowforge/issues/1747

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [x] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: test/customizations-smtp.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+forge:
```
Comment (ZJvandeWeg) [inline test/customizations-smtp.yml:1]: Not quite sure what we need this file for?

**Comment by ZJvandeWeg** [comment]
This looks good, just not sure why the validator file was staged?

**Comment by ZJvandeWeg** [comment]
@hardillb feel free to self merge if the file should be staged.

File: test/customizations-smtp.yml (hunk 1 above)
Comment (hardillb) [reply test/customizations-smtp.yml:1]: It's some extra tests, I'll remove it

**Pull request merged**